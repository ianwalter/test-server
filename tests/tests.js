const { test } = require('@ianwalter/bff')
const r2 = require('r2')
const createTestServer = require('..')

test('server created', async ({ expect }) => {
  const server = await createTestServer()
  expect(server.port).toBeGreaterThan(0)
  expect(server.url).toBeTruthy()
  await server.close()
})

test('request handler', async ({ expect }) => {
  const server = await createTestServer()
  const msg = 'Nobody Lost, Nobody Found'
  server.use(ctx => (ctx.body = msg))
  const body = await r2(server.url).text
  expect(body).toBe(msg)
  await server.close()
})

test('json response', async ({ expect }) => {
  const server = await createTestServer()
  server.use(ctx => (ctx.body = { name: 'Out There On the Ice' }))
  const body = await r2(server.url).text
  expect(body).toMatchSnapshot()
  await server.close()
})

test('json request', async ({ expect }) => {
  const server = await createTestServer()
  server.use(ctx => (ctx.body = ctx.request.body))
  const json = { name: 'When Am I Gonna Lose You' }
  const body = await r2.post(server.url, { json }).json
  expect(body).toEqual(json)
  await server.close()
})

// test('cors', async (t, page) => {
//   const server = await createTestServer()
//   server.use(ctx => (ctx.body = 'Moments'))
//   const result = await page.evaluate(
//     url => window.fetch(url).then(response => response.text()),
//     server.url
//   )
//   t.is(result, 'Moments')
// })

test('error', async ({ pass }) => {
  const server = await createTestServer()
  server.use(() => new Promise((resolve, reject) => reject(new Error('Nooo!'))))
  try {
    await r2(server.url).json
  } catch (err) {
    pass()
  } finally {
    await server.close()
  }
})
