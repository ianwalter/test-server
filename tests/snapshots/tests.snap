// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`json response 1`] = `
"{
  \\"name\\": \\"Out There On the Ice\\"
}"
`;
